### PR TITLE
Less globals and add cpu smi for consistency

### PR DESCRIPTION
**Pull request by Delaunay**

### CODE REVIEW

**Comment by Delaunay** [comment]
@breuleux quand tu as le temps

**Comment by breuleux** [comment]
@Delaunay Quel problème est-ce que ça règle exactement?

**Comment by Delaunay** [comment]
It was possible to call `get_visible_devices` before the backend was initialized.
This PR make it harder to query an uninitialized backend by making all backend query go through the SMI instance instead.
The only public method to retrieve the SMI instance will make sure the backend is indeed initialized 

Instance of the issue here
https://github.com/mila-iqia/milabench/actions/runs/4873967932/jobs/8694286168?pr=121#step:9:103

File: tests/test_instruments.py [hunk 1]
```diff
@@ -63,11 +63,12 @@ def sync():
...
 
 def test_gpu_backend():
-    assert list(sorted(get_backends())) == ["cuda", "rocm"]
+    assert list(sorted(get_backends())) == ["cpu", "cuda", "rocm"]
 
 
 def test_gpu_info_cpu():
-    assert get_gpu_info(arch="cpu") == {"arch": "cpu", "gpus": {}}
+    smi = select_backend()
+    assert get_gpu_info(smi) == {"arch": "cpu", "gpus": {}}
```
Comment (breuleux) [inline tests/test_instruments.py]: Could we keep the original interface for `get_gpu_info`? It is a bit simpler and it is used in multiple places in milabench.

Comment (Delaunay) [reply tests/test_instruments.py]: done

**Review by breuleux — APPROVED** [review]

**Pull request merged**